src/Application/UI/Presenter.php

```diff
@@ -1055,19 +1058,21 @@ public static function argsToParams($class, $method, & $args, $supplemental = ar
 
 	/**
 	 * Invalid link handler. Descendant can override this method to change default behaviour.
+	 * @param InvalidLinkException $e
 	 * @return string
 	 * @throws InvalidLinkException
 	 */
 	protected function handleInvalidLink(InvalidLinkException $e)
 	{
-		if ($this->invalidLinkMode === self::INVALID_LINK_SILENT) {
-			return '#';
-
-		} elseif ($this->invalidLinkMode === self::INVALID_LINK_WARNING) {
-			return '#error: ' . $e->getMessage();
-
-		} else { // self::INVALID_LINK_EXCEPTION
-			throw $e;
+		switch($this->invalidLinkMode) {
+			case self::INVALID_LINK_WARNING:
```

Comment:
Exception should not be dropped, it should produce at least a warning. Untill it's removed few versions later.